apps/emqx/include/emqx_lsr.hrl

```diff
@@ -0,0 +1,20 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2025 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+-ifndef(EMQX_LSR_HRL).
+-define(EMQX_LSR_HRL, true).
+
+-define(lsr_err_max_retries, lsr_err_max_retries).
+-define(lsr_err_channel_outdated, lsr_err_channel_outdated).
+-define(lsr_err_restart_takeover, lsr_err_restart_takeover).
+
+-record(lsr_channel, {
+    id :: lsr_session_id(),
+    pid :: pid() | '$1',
+    vsn :: integer() | undefined | '$2'
+}).
```

Comment:
Nit: when a record is used in match, usually it's better to avoid in-place field type declarations. This code can be changed as following:

emqx_lsr.hrl:
```
-record(lsr_channel, {id, pid, vsn}).
```

emqx_cm.erl

```
-type lsr_channel() :: #lsr_channel{id :: lsr_session_id(), pid :: pid(), ...}.
```
```

Reply:
good point!  but I prefer to have types in place for readability. I will add a note for these values used for match spec.